api/src/test/java/org/pmiops/workbench/firecloud/FireCloudServiceImplTest.java

```diff
@@ -318,6 +318,25 @@ public void isUserCompliantWithTerraToS_false()
     verifyNoInteractions(firecloudTermsOfServiceApi);
   }
 
+  @Test
+  public void isUserCompliantWithTerraToS_404()
+      throws org.broadinstitute.dsde.workbench.client.sam.ApiException {
+    when(termsOfServiceApi.userTermsOfServiceGetSelf())
+        .thenThrow(new org.broadinstitute.dsde.workbench.client.sam.ApiException(404, "Not found"));
+    assertThat(service.isUserCompliantWithTerraToS()).isFalse();
+
+    verify(termsOfServiceApi).userTermsOfServiceGetSelf();
+    verifyNoInteractions(firecloudTermsOfServiceApi);
```

Comment:
What is the difference between these two ToS apis?

It seems like one is Sam and one is general Terra. Do we use these two independently?

Reply:
The Terra ToS APIs are "old-style" which we are migrating away from in the epic.  The Sam ToS APIs are the destination we're migrating to.

Reply:
this epic -> https://precisionmedicineinitiative.atlassian.net/browse/RW-11416